### PR TITLE
Fix sdpfa_bwd to not assume the presence of DIDs

### PR DESCRIPTION
**Pull request by wujingyue**
Similar to https://github.com/NVIDIA/Fuser/pull/3116.

### CODE REVIEW

**Comment by wujingyue** [comment]
!build

File: csrc/ops/composite.cpp [hunk 1]
```diff
@@ -546,48 +560,39 @@ SdpfaBwdResult sdpfa_bwd(
...
-  NVF_CHECK(
-      query->dtype() == key->dtype() && query->dtype() == value->dtype(),
-      "Expected query, key, and value to have the same dtype but got: ",
-      query->dtype(),
-      " ",
-      key->dtype(),
-      " ,and ",
-      value->dtype());
-
+  checkAllEqual(
```
Comment (wujingyue) [inline csrc/ops/composite.cpp:563]: This and the checkAllEqual at line 576 will harden the checks. Currently, the code checks only Q, K and V.

File: csrc/ops/composite.cpp [hunk 2]
```diff
@@ -534,6 +534,20 @@ SdpfaFwdResult sdpfa_fwd(
   return {output, log_sumexp, philox_seed, philox_offset};
 }
 
+namespace {
+template <typename T>
+void checkAllEqual(std::initializer_list<T> elements) {
```
Comment (Priya2698) [inline csrc/ops/composite.cpp:539]: Nice template, we can propagate this to `sdpfa_fwd` above as well (does not have to be this PR)

**Review by Priya2698 — APPROVED** [review]
It should be possible to use https://github.com/NVIDIA/Fuser/blob/aec6797d6386cbfe02c1b5405b7e0a9439915295/tests/python/utils.py#L387 here as noted in comment [here](https://github.com/NVIDIA/Fuser/pull/3116#discussion_r1790715297). See `test_python_frontend.py` for usage example.
I don't think this switch is a blocker for this PR if you decide to go this route and can be done separately.

LGTM once the CI error is resolved.

**Comment by wujingyue** [comment]
> It should be possible to use
> 
> https://github.com/NVIDIA/Fuser/blob/aec6797d6386cbfe02c1b5405b7e0a9439915295/tests/python/utils.py#L387

Alas, this doesn't quite work out, and I created https://github.com/NVIDIA/Fuser/issues/3214.

File: csrc/ops/composite.cpp (hunk 2 above)
Comment (wujingyue) [reply csrc/ops/composite.cpp]: Done!

**Comment by wujingyue** [comment]
!build

**Pull request merged**